### PR TITLE
You might find it useble to add SSL configuration into your role.  

### PR DESCRIPTION
**Pull request by riversy**
First of all, Thank you for your book Ansible for DevOps :)
I'm reading your book and create infrastructure for our first SaaS application. And using Asnible first time in my live. 
I'm just found that this role doesn't support SSL. You might find it useble to add this piece into your role.

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -8,7 +8,9 @@ haproxy_group: haproxy
 haproxy_frontend_name: 'hafrontend'
 haproxy_frontend_bind_address: '*'
 haproxy_frontend_port: 80
+haproxy_frontend_ssl_port: 443
 haproxy_frontend_mode: 'http'
+haproxy_ssl_pem: false
```
Comment (geerlingguy) [inline defaults/main.yml]: For the default, please change this to an empty string.

Also, can you add a line in the README.md file documenting this variable so first-time users of the role know to add a path to their pem file on the local/Ansible workstation?

File: tasks/main.yml [hunk 2]
```diff
@@ -25,6 +25,14 @@
   set_fact:
     haproxy_version: "{{ '1.5' if '1.5.' in haproxy_version_result.stdout else '1.4' }}"
 
+- name: Copy SSL certificate
+  copy:
+    src: "{{ haproxy_ssl_pem }}"
+    dest: /etc/haproxy/server.pem
+    mode: 0644
+  notify: restart haproxy
+  when: haproxy_ssl_pem != false
```
Comment (geerlingguy) [inline tasks/main.yml]: Change `false` to `''` to check against empty string after changing the variable default to an empty string.

File: defaults/main.yml [hunk 3]
```diff
@@ -8,12 +8,15 @@ haproxy_group: haproxy
 haproxy_frontend_name: 'hafrontend'
 haproxy_frontend_bind_address: '*'
 haproxy_frontend_port: 80
+haproxy_frontend_ssl_port: 443
 haproxy_frontend_mode: 'http'
+haproxy_ssl_pem: ''
```
Comment (geerlingguy) [inline defaults/main.yml:13]: Can you document this new set of variables inside README.md? That way users will know what they are and what's necessary to get SSL working (e.g. a valid .pem).

Comment (riversy) [inline defaults/main.yml:13]: Hi @geerlingguy, 
Wasn't able to do it due to overloading. Will done today. Thanx for your patience.

File: templates/haproxy.cfg.j2 [hunk 4]
```diff
@@ -37,15 +37,26 @@ defaults
 
 frontend {{ haproxy_frontend_name }}
     bind {{ haproxy_frontend_bind_address }}:{{ haproxy_frontend_port }}
+    {% if haproxy_ssl_pem %}
+    bind {{ haproxy_frontend_bind_address }}:{{ haproxy_frontend_ssl_port }} ssl crt /etc/haproxy/server.pem
```
Comment (geerlingguy) [inline templates/haproxy.cfg.j2:41]: Does `/etc/haproxy/server.pem` need to be configurable? Does it work cross-platform (e.g. RHEL and Ubuntu)?

**Comment by geerlingguy** [comment]
Just following up; thanks again for the PR, reviewed twice, if you get some time to incorporate the feedback/answer the questions, I think this would make a great addition to the role!

**Comment by clwells** [comment]
PR'd with changes requested here: https://github.com/riversy/ansible-role-haproxy/pull/1

Also, it looks like /etc/haproxy exists on both Ubuntu/RHEL so the server.pem living there should be fine.

**Comment by 030** [comment]
Could you indicate when this will be implemented?

**Comment by thomasjungblut** [comment]
@riversy @geerlingguy do you guys want to merge it at some point? I'd rather not vendor this whole thing with upstream merges manually to get it working.

**Pull request closed without merging**